### ISSUE TITLE
Call default rake task inside migration file succeeded but raise errors.

### ISSUE THREAD

**Issue by trungtq-0433** [rails/rails#39331]
### Steps to reproduce
I have a migration file, and run `rails db:migrate`
```
class OriginalDatabaseMigration < ActiveRecord::Migration[6.0]
  def up
    Rake::Task["db:structure:load"].invoke
  end
end
```
### Expected behavior
Loaded database structure and no raise errors.

### Actual behavior
```
== 1 OriginalDatabaseMigration: migrating =====================================
mysql: [Warning] Using a password on the command line interface can be insecure.
mysql: [Warning] Using a password on the command line interface can be insecure.
== 1 OriginalDatabaseMigration: migrated (0.2575s) ============================

rake aborted!
Mysql2::Error: MySQL client is not connected

Tasks: TOP => db:migrate:up
(See full trace by running task with --trace)
```

### System configuration
**Rails version**: 6.0.2.1

**Ruby version**: 2.6.5p114

**Comment by eileencodes** [comment]
Hi @trungtq-0433 thanks for the report. Can you explain more your use case that needs to support this? We have a rake task you can call for loading the structure and I think it's very uncommon to call the task inside a migration like this. Did this work in previous Rails versions?

**Comment by eileencodes** [comment]
Can you also try with 6.0.3.1? This error is similar to another report and may be fixed in the most recent release of 6.0.3.

**Comment by eileencodes** [comment]
I'm going to close this out since there's been no response. If you're still having issues @trungtq-0433 after testing with the 6.0.3 release, let me know and I'll reopen.
